glue/src/main/scala/com/gu/tableversions/glue/GlueMetastore.scala

```diff
@@ -112,18 +115,22 @@ class GlueMetastore[F[_]](glue: AWSGlue)(implicit F: Sync[F]) extends Metastore[
       F.delay(glue.updatePartition(updatePartitionRequest)).void
     }
 
-    versionedPartitionLocation(table, partition, version).flatMap(location => updatePartition(location).void)
+    versionedPartitionStorageDescriptor(table, partition, version).flatMap(location => updatePartition(location).void)
   }
 
-  private def versionedPartitionLocation(table: TableName, partition: Partition, version: Version): F[URI] =
+  private def versionedPartitionStorageDescriptor(
+      table: TableName,
+      partition: Partition,
+      version: Version): F[StorageDescriptor] =
     for {
       gluetable <- getGlueTable(table)
       tableLocation = findTableLocation(gluetable)
       partitionLocation = partition.resolvePath(tableLocation)
       versionedPartitionLocation = VersionPaths.pathFor(partitionLocation, version)
-    } yield versionedPartitionLocation
+    } yield extractFormatParams(gluetable.getStorageDescriptor).withLocation(versionedPartitionLocation.toString)
```

Comment:
it would probably be more technically correct to get the existing partition storage descriptor and keep the existing parameters while updating just the location. But that would be an extra api call and the partitions are created by cloning the table storage descriptor so the end result should be the same unless the partition is created or updated bypassing this GlueMetastore